File: emis-backend/src/main/java/com/emis/fee/FeeService.java
package com.emis.fee;

import java.util.List;

import com.emis.fee.dto.FeeRequest;
import com.emis.fee.dto.FeeResponse;
import com.emis.fee.dto.UpdateFeeRequest;

public interface FeeService {

    FeeResponse addFee(FeeRequest request);

    FeeResponse getFeeByStudent(Long studentId);

    List<FeeResponse> getAllFees();

    FeeResponse updateFee(Long feeId, UpdateFeeRequest request);

    void deleteFee(Long feeId);
}

File: emis-frontend/src/services/feeService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/fees";

export const getAllFees = () => {
  return axios.get(API_URL);
};

export const addFee = (feeData) => {
  return axios.post(API_URL, feeData);
};

export const updateFee = (feeId, data) => {
  return axios.put(`${API_URL}/${feeId}`, data);
};

export const getFeeByStudent = (studentId) => {
  return axios.get(`${API_URL}/student/${studentId}`);
};

export const deleteFee = (feeId) => {
  return axios.delete(`${API_URL}/${feeId}`);
};
